core/local/local.go

```diff
@@ -224,6 +224,19 @@ func (e *ExecutionScheduler) initVUsConcurrently(
 func (e *ExecutionScheduler) Init(ctx context.Context, samplesOut chan<- stats.SampleContainer) error {
 	logger := e.logger.WithField("phase", "local-execution-scheduler-init")
 
+	// Initialize each executor
```

Comment:
Hmm I intentionally left it here, since initializing VUs is very costly compared to executors, and there's no point in initializing VUs if the exec init fails. Though it's a moot point since exec `Init()` is rarely used (only in CAR and SI currently, which don't return an error).

> just conceptually, since executors depend on VUs and not the other way around

Is it weird that I see it the other way around? :D VUs are at the bottom of the dependency chain, so to speak (Engine -> ExecutionScheduler -> Executors -> VUs). That is, without executors there'd be nothing to actually run VUs. This distinction is a bit blurry since ExecutionScheduler kind of deals with executors, VUs and Runner, but conceptually I see VUs depending on executors.

But no problem, I'll move this back to where it was.

Reply:
> VUs are at the bottom of the dependency chain, so to speak (Engine -> ExecutionScheduler -> Executors -> VUs)

hmm fair enough, I can see the logic, so I retract my nitpicking :sweat_smile:

Reply:
No prob, I still reverted it in 60c3c099.